src/mon/Paxos.cc

```diff
@@ -1203,6 +1205,14 @@ void Paxos::reset_lease_timeout()
       }));
 }
 
+void Paxos::cancel_lease_timeout()
+{
+  lease_expire = utime_t();  // cancel lease
+  dout(20) << __func__ << " - unsetting timeout event" << dendl;
+  if (lease_timeout_event)
+    mon->timer.cancel_event(lease_timeout_event);
```

Comment:
i don't think we should reset cancel the lease timeout event. and it's scaring that `lease_timeout_event` is not scheduled right away. the point of setting the `lease_timeout_event` is to ensure that we have a responsive leader. if the leader fails to update its peons the lease message in time. we have no reason to tolerate it.

in your case, i think you might want to move the application hammering your disk away from monitor, or use SSD/NVMe device for the monstore if not yet.

Reply:
it's for invalidating the current release. in other words, marking the current updated.

Reply:
i see! in proposing, the lease was set invalidate, then mon can not be readed or writed.
the comment “ //cancel lease" really make me confused.

i think we should modify the s/cancel lease/invalidate lease/

Reply:
"cancel lease" makes more sense to me. as lease is a contract. cancel in this context is more specific and accurate. "invalidate" is too general.